packages/bitcoin-ddk-provider/lib/BitcoinDdkProvider.ts

```diff
@@ -328,25 +328,23 @@ export default class BitcoinDdkProvider extends Provider {
   ): Promise<Input[]> {
     if (amounts.length === 0) return [];
 
-    // For "none" mode, use exactly the provided inputs
-    if (supplementation === InputSupplementationMode.None) {
-      return fixedInputs;
-    }
-
-    // For "required" and "optional" modes, attempt supplementation
+    // Convert to UTXOs
     const fixedUtxos = fixedInputs.map((input) => input.toUtxo());
 
     try {
       const inputsForAmount: InputsForDualAmountResponse = await this.getMethod(
         'getInputsForDualFunding',
-      )(amounts, feeRatePerVb, fixedUtxos);
+      )(amounts, feeRatePerVb, fixedUtxos, supplementation);
 
       // Convert UTXO objects to Input class instances
       return inputsForAmount.inputs.map((utxo) => Input.fromUTXO(utxo));
     } catch (e) {
       const errorMessage = e instanceof Error ? e.message : 'Unknown error';
 
-      if (supplementation === InputSupplementationMode.Required) {
+      if (
+        supplementation === InputSupplementationMode.Required ||
+        supplementation === InputSupplementationMode.None
+      ) {
         throw Error(
           `Not enough balance GetInputsForAmountWithMode. Error: ${errorMessage}`,
```

Comment:
The error isn't really not enough balance now, it's invalid supplementation mode